sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestFramework/VerifierTest.cs

```diff
@@ -0,0 +1,80 @@
+﻿/*
+ * SonarAnalyzer for .NET
+ * Copyright (C) 2015-2020 SonarSource SA
+ * mailto: contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+extern alias csharp;
+using FluentAssertions;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using SonarAnalyzer.Common;
+using SonarAnalyzer.Helpers;
+using System.Collections.Immutable;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CodeActions;
+
+namespace SonarAnalyzer.UnitTest.TestFramework
+{
+    [TestClass]
+    public class VerifierTest
+    {
+        [TestMethod]
+        public void VerifyCodeFix_WithDuplicateIssues()
+        {
+            const string filename = @"TestCases\VerifyCodeFix.Empty.cs";
+            System.Action a = () => { Verifier.VerifyCodeFix(filename, filename, new TestDuplicateLocationRule(), new TestDuplicateLocationRuleCodeFix()); };
+            a.Should().NotThrow();
+        }
+
+        [Microsoft.CodeAnalysis.Diagnostics.DiagnosticAnalyzer(LanguageNames.CSharp)]
+        private class TestDuplicateLocationRule : SonarDiagnosticAnalyzer
+        {
+            public const string DiagnosticId = "Test42";
+            private DiagnosticDescriptor rule = new DiagnosticDescriptor(DiagnosticId, "Title", "Message", "Category", DiagnosticSeverity.Warning, true, null, null, new[] { "MainSourceScope" });
```

Comment:
Since **DiagnosticDescriptor** ctor accepts params, `new[] { "MainSourceScope" }` can be replaced with `"MainSourceScope"`